lib/rubocop/cop/rspec/capybara/specific_matcher.rb

```diff
@@ -56,7 +79,31 @@ def specific_matcher(arg)
           end
 
           def acceptable_pattern?(arg)
-            arg.match?(/\[.+=\w+\]/) || arg.match?(/[ >,+]/)
+            arg.match?(/[ >,+]/)
+          end
+
+          def specific_matcher_option?(arg, matcher)
+            # If `button[foo-bar_baz=foo][bar][baz]`:
+            # extract ["foo-bar_baz", "bar", "baz"]
+            attributes = arg.scan(/\[(.*?)\]/).flatten
```

Comment:
If we want to go all in on regex:
```suggestion
            attributes = arg.scan(/\[(.*?)(?:=.*?)?\]/).flatten
```
Then we can delete the `.split('=').first` in line 92.

Reply:
It is clearly better that way. Thank you so much.